src/backend/marsha/core/models/video.py

```diff
@@ -469,3 +471,84 @@ class Meta:
         ]
 
         verbose_name = _("live registration")
+
+
+class LivePairing(BaseModel):
+    """Model representing a live pairing."""
+
+    RESOURCE_NAME = "livepairings"
+
+    secret = models.CharField(
+        max_length=6,
+        unique=True,
+        verbose_name=_("secret"),
+        help_text=_("live pairing secret"),
+    )
+
+    expires_at = models.DateTimeField()
+
+    video = models.OneToOneField(
+        to="Video",
+        related_name="live_pairing",
+        verbose_name=_("video"),
+        help_text=_("live pairing video"),
+        # LivePairing is (soft-)deleted if video is (soft-)deleted
+        on_delete=models.CASCADE,
+    )
+
+    class Meta:
+        """Options for the ``LivePairing`` model."""
+
+        db_table = "live_pairing"
+        ordering = ["expires_at"]
+        verbose_name = _("live pairing")
+        verbose_name_plural = _("live pairings")
+
+    def __str__(self):
+        """Get the string representation of an instance."""
+        result = f"{self.video.title}"
+        if self.deleted:
+            result = _("{:s} [deleted]").format(result)
+        return result
+
+    @classmethod
+    def secret_generator(cls, existing_secrets):
+        """Generates a random 6 digit string."""
+        # Generate a list of all possible 6 digits strings
+        all_secrets = (str(i).zfill(6) for i in range(0, 1_000_000))
+
+        # Difference between all and existing secrets
+        available_secrets = set(all_secrets) - set(existing_secrets)
+
+        # Return one of available secrets
+        return secrets.choice(list(available_secrets))
+
+    @property
+    def expired(self):
+        """Returns true if secret has expired."""
+        return self.expires_at < timezone.now()
+
+    def generate_secret(self):
+        """Stores generated secret and expiration date."""
+        existing_secrets = LivePairing.objects.values_list("secret", flat=True)
```

Comment:
I would say this is overkill and can cause race conditions. A try/except loop may be appropriate here?
```
for _ in range(2):
    self.secret = secrets.randint(1000000)
    try:
       self.save()
    except ...:
        pass
    else:
        break
```
If we fall on an existing secret twice in a row, it is a sign that we have too many challenges ongoing? Do you think we need to formally limit the number of challenges being run in parallel in order to ensure a very low probability of finding a secret by chance?

Reply:
Sorry, but I don't get which race condition would be caused, and in which way the approach I propose is overkill…
Can you elaborate on this, please ?

Reply:
Because two concurrent requests could generate a new code that would be the same... So the second one would fail upon db transaction commit. We also need to detect if too may pairing challenges are open simultaneously as it would kill the security of the process?

Reply:
Regarding the current code state, and discussions we had, is your comment still relevant ?